architecture/2024-08-controller-component-threat-model.md

```diff
@@ -0,0 +1,241 @@
+# Title
+
+- **Author**: ytimocin
+
+## Overview
+
+This document provides a threat model for the Radius Controller component. It identifies potential security threats to this critical part of Radius and suggests possible mitigations. The document includes an analysis of the system, its assets, identified threats, and recommended security measures to protect the system.
+
+The Radius Controller component monitors changes (create, update, delete) in the definitions of Recipe and Deployment resources. Based on these changes, the appropriate controller takes the necessary actions. Below, you will find detailed information about the key parts of the Radius controllers and the objects they manage.
+
+## Terms and Definitions
+
+| Term                  | Definition                                                                                                                                                                                  |
+| --------------------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| Admission Controllers | An admission controller is a piece of code that intercepts requests to the Kubernetes API server prior to persistence of the object, but after the request is authenticated and authorized. |
+| mTLS                  | Mutual Transport Layer Security (mTLS) allows two parties to authenticate each other during the initial connection of an SSL/TLS handshake.                                                 |
+
+## System Description
```

Comment:
I have a meta-question - when we change these components, this doc will be out of date. Is the expectation that we update these on feature implementations or do we just write a new one every once in a while?

Reply:
Good question. We can talk about this. I believe that whenever there is an important update, we need to go and update this document too.

For example, if we update the logic of the existing validating webhook or if we add a mutating webhook, we should update the threat model too.